File: ts/01(数组)移除元素/27.移除元素.ts
/*
 * @lc app=leetcode.cn id=27 lang=typescript
 *
 * [27] 移除元素
 */

// @lc code=start
function removeElement(nums: number[], val: number): number {
  let low = 0
  let high = 0
  while (high < nums.length) {
    //如果high指针指向的值符合target,将最前面的值依次变为target
    if (nums[high] !== val) nums[low++] = nums[high]
    high++
  }
  nums[low] = val
  console.log(nums)
  return low
}
// @lc code=end

console.log(removeElement([1, 2, 2, 3, 4, 3], 3))

File: c/src/Array/27.移除元素.c
/*
 * @lc app=leetcode.cn id=27 lang=c
 *
 * [27] 移除元素
 */

// @lc code=start

int removeElement(int *nums, int numsSize, int val) {
  int slow = 0;
  for (int fast = 0; fast < numsSize; fast++) {
    if (nums[fast] != val) {
      nums[slow++] = nums[fast];
    }
  }
  return slow;
}
// @lc code=end
